internal/backend/vars/vars.go

```diff
@@ -46,6 +63,25 @@ func (v *Vars) Set(ctx context.Context, vs ...sdktypes.Var) error {
 
 			vs[i] = va.SetValue(key)
 		}
+
+		if v.cfg.MaxValueSize != 0 && len(va.Value()) > v.cfg.MaxValueSize {
+			return fmt.Errorf("%w: value size %d exceeds max value size %d", sdkerrors.ErrLimitExceeded, len(va.Value()), v.cfg.MaxValueSize)
+		}
+
+		scids[va.ScopeID()] = true
+	}
+
+	if maxN := v.cfg.MaxNumVarsPerScope; maxN != 0 && len(vs) > maxN {
+		for scid := range scids {
+			n, err := v.db.CountVars(ctx, scid)
```

Comment:
does this do N db queries to where N == Number of vars you try to set ?
if so, please refactor to query once in a transaction,
otherwise, I'm probably missing something

Reply:
consider maybe change the vars api to work in a single scope per call

Reply:
now with transaction